packages/backend-for-frontend/src/services/clientService.ts

```diff
@@ -333,14 +333,19 @@ async function enhanceClient(
   client: authorizationApi.Client,
   ctx: WithLogger<BffAppContext>
 ): Promise<bffApi.Client> {
-  const consumer = await apiClients.tenantProcessClient.tenant.getTenant({
-    params: { id: client.consumerId },
-    headers: ctx.headers,
-  });
-
-  const purposes = await Promise.all(
-    client.purposes.map((p) => enhancePurpose(apiClients, p, ctx))
-  );
+  const [consumer, admin, purposes] = await Promise.all([
+    await apiClients.tenantProcessClient.tenant.getTenant({
+      params: { id: client.consumerId },
+      headers: ctx.headers,
+    }),
+    client.adminId
+      ? await apiClients.tenantProcessClient.tenant.getTenant({
+          params: { id: client.adminId },
+          headers: ctx.headers,
+        })
+      : undefined,
+    Promise.all(client.purposes.map((p) => enhancePurpose(apiClients, p, ctx))),
```

Comment:
```suggestion
  const [consumer, admin, ...purposes] = await Promise.all([
    apiClients.tenantProcessClient.tenant.getTenant({
      params: { id: client.consumerId },
      headers: ctx.headers,
    }),
    client.adminId
      ? apiClients.tenantProcessClient.tenant.getTenant({
          params: { id: client.adminId },
          headers: ctx.headers,
        })
      : Promise.resolve(undefined),
    ...client.purposes.map((p) => enhancePurpose(apiClients, p, ctx)),
```
Inside a `Promise.all` you have to return promises, so we don't have to await and instead of returning undefined we return `Promise.resolve(undefined)`; the second `Promise.all` is not need since it can be handled by the parent one